src/lib.rs

```diff
@@ -212,6 +214,49 @@ impl FromStr for Message {
     }
 }
 
+#[cfg(feature = "serde")]
+impl Serialize for Message {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        serializer.serialize_str(self.to_string().as_str())
+    }
+}
+
+#[cfg(feature = "serde")]
+struct MessageVisitor;
+
+#[cfg(feature = "serde")]
+impl<'de> Visitor<'de> for MessageVisitor {
+    type Value = Message;
+
+    fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        formatter.write_str("an EIP-4361 formatted message")
+    }
+
+    fn visit_str<E>(self, value: &str) -> Result<Self::Value, E>
+    where
+        E: de::Error,
+    {
+        let message = match Message::from_str(value) {
+            Ok(message) => Ok(message),
+            Err(error) => Err(E::custom(format!("error parsing message: {}", error))),
+        };
+        message
+    }
+}
+
+#[cfg(feature = "serde")]
+impl<'de> Deserialize<'de> for Message {
+    fn deserialize<D>(deserializer: D) -> Result<Message, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        deserializer.deserialize_str(MessageVisitor)
+    }
+}
```

Comment:
For types that implement `FromStr` I usually do:

```rust
impl<'de> Deserialize<'de> for Message {
    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
        where D: Deserializer<'de>
    {
        String::deserialize(deserializer)?
            .parse().map_err(DeserializerError::custom)
    }
}
```

Without writing a Visitor. Perhaps this approach can be used here too.

Reply:
Please suggest the alternative if you'd like to see it, otherwise this PR is fully baked/working for me. This was what the `serde` docs recommended.

Reply:
Personally I prefer the custom visitor and the error message detail it provides.

Reply:
Agreed. Implementing a visitor gives a much better error message.

Reply:
I don't quite understand what you mean by that. You can similarly adjust error message by changing `map_err` in the example I provided: 

```rust
String::deserialize(deserializer)?.parse()
    .map_err(|error| DeserializerError::custom(format!("error parsing message: {}", error)))
```

Reply:
That's only for the error message from `Message::from_str`. The implementation of `Visitor::expecting` improves the serde error:
```
Error("invalid type: map, expected an EIP-4361 formatted message", line: 1, column: 0)
```